lib/floe/workflow/error_matcher.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    # Methods for common error handling
+    #
+    # @note Included class required to have `error_equals` available from the payload.
+    module ErrorMatcher
+      # @param [String] error the error thrown
+      def match_error?(error)
+        return false if error == "States.Runtime"
+        return true if error_equals.include?("States.ALL")
```

Comment:
Do we need to check that the author set error_equals correctly?  If it was unset, then I think this is nil, and if not set to an Array, this is a string include check?  (I think the old code also failed, but it was more subtle).

Should we add validation of the error equals payload?

Reply:
old code was:

```ruby
(r.error_equals & [error, "States.ALL"]).any?

nil & [1,2]
# => nil
```

So yea, it silently failed.

Reply:
Added validation so we catch a retry with an empty, missing, or non-array ErrorEquals